src/cascadia/TerminalApp/WindowsPackageManagerFactory.h

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+// Module Name:
+// - WindowsPackageManagerFactory.h
+//
+// Abstract:
+// - This factory is designed to create production-level instances of WinGet objects.
+//   Elevated sessions require manual activation of WinGet objects and are not currently supported,
+//   while non-elevated sessions can use the standard WinRT activation system.
+// Author:
+// - Carlos Zamora (carlos-zamora) 23-Jul-2024
+
+#pragma once
+
+#include "../../types/inc/utils.hpp"
+#include <combaseapi.h>
+#include <wil/result_macros.h>
+#include <wil/win32_helpers.h>
+#include <type_traits>
+
+#include <winrt/Microsoft.Management.Deployment.h>
+
+namespace winrt::TerminalApp::implementation
+{
+    class WindowsPackageManagerFactory
+    {
+    public:
+        static WindowsPackageManagerFactory& Instance()
+        {
+            static WindowsPackageManagerFactory instance;
+            return instance;
+        }
+
+        // Delete copy constructor and assignment operator
+        WindowsPackageManagerFactory(const WindowsPackageManagerFactory&) = delete;
+        WindowsPackageManagerFactory& operator=(const WindowsPackageManagerFactory&) = delete;
+
+        winrt::Microsoft::Management::Deployment::PackageManager CreatePackageManager()
+        {
+            static winrt::guid clsid{ "C53A4F16-787E-42A4-B304-29EFFB4BF597" };
+            return winrt::create_instance<winrt::Microsoft::Management::Deployment::PackageManager>(clsid, CLSCTX_ALL);
+        }
+
+        winrt::Microsoft::Management::Deployment::FindPackagesOptions CreateFindPackagesOptions()
+        {
+            static winrt::guid clsid{ "572DED96-9C60-4526-8F92-EE7D91D38C1A" };

```

Comment:
`static` on its own declares a mutable variable. These can all be constants. This allows the linker to put the values into the read-only memory parts of the resulting binary, instead of having to copy them out of the binary into mutable memory on startup. The most reliable way to achieve that is with `static constexpr`. You may have to use the numeric constructor instead of the string one though.